packages/next-swc/crates/next-core/src/next_client/context.rs

```diff
@@ -372,12 +372,39 @@ pub async fn get_client_runtime_entries(
         );
     }
 
-    // It's important that React Refresh come before the regular bootstrap file,
-    // because the bootstrap contains JSX which requires Refresh's global
-    // functions to be available.
-    if let Some(request) = enable_react_refresh {
-        runtime_entries.push(RuntimeEntry::Request(request, project_root.join("_")).cell())
-    };
+    match mode {
+        NextMode::Development => {
+            let resolve_options_context = get_client_resolve_options_context(
+                project_root,
+                ty,
+                mode,
+                next_config,
+                execution_context,
+            );
+            let enable_react_refresh =
+                assert_can_resolve_react_refresh(project_root, resolve_options_context)
+                    .await?
+                    .as_request();
+
+            // It's important that React Refresh come before the regular bootstrap file,
+            // because the bootstrap contains JSX which requires Refresh's global
+            // functions to be available.
+            if let Some(request) = enable_react_refresh {
+                runtime_entries.push(RuntimeEntry::Request(request, project_root.join("_")).cell())
+            };
+        }
+        NextMode::Build => {
```

Comment:
There's no way to run into this branch (yet), but this is necessary for the next PRs.